src/app/src/api/tags.js

```diff
@@ -0,0 +1,26 @@
+const TAGS = {
+    SUBMISSION: 'submission',
+
+    LIST_ID: 'list',
+};
+
+export function getListTagProvider(tag) {
+    return results => [
+        ...results.map(({ id }) => ({ tag, id })),
+        { tag, id: TAGS.LIST_ID },
+    ];
+}
+
+export function getSingleItemProvider(tag) {
+    return (result, error, id) => ({ tag, id });
+}
+
+export function getNewItemTagInvalidator(tag) {
+    return () => ({ tag, id: TAGS.LIST_ID });
+}
+
+export function getUpdateItemTagInvalidator(tag) {
+    return (result, error, id) => ({ tag, id });
+}
+
```

Comment:
In my past experience with RTK Query, managing the API tags for cache invalidation can be a pain point. I added these helper functions mostly as a way of guiding future implementation of tags. We needn't get too attached to this pattern though. It might turn out they are not flexible enough for some parts of our data schema. 

Tags are the recommended way to go from RTK Query, but it is possible to manually update the cache. More info can be found here: https://redux-toolkit.js.org/rtk-query/usage/manual-cache-updates.